pyclient/pymtt.py

```diff
@@ -144,87 +144,44 @@
-            else:
-                print("The TestDef.py file was not found in the standard location")
-                print("under the specified base directory at",chkdir)
-                sys.exit(1)
-elif os.path.exists("TestDef.py"):
-    # the class file is local to us, so use it
-    basedir = "./"
-    topdir = basedir
-elif os.path.exists("pylib"):
-    # we appear to be in the MTT home directory
-    basedir = os.path.join("./", "pylib", "System")
-    topdir = os.path.join("./", "pylib")
-    if not os.path.exists(basedir) or not os.path.isdir(basedir):
-        print("The local directory",basedir,"doesn't exist")
-        sys.exit(1)
-    if not os.path.exists(os.path.join(basedir, "TestDef.py")):
-        print("The TestDef.py file was not found in the standard location")
-        print("under the specified base directory at",basedir)
-        sys.exit(1)
-else:
-    try:
-        # try that location
-        basedir = os.path.join(os.environ['MTT_HOME'], "pylib", "System")
-        if not os.path.exists(basedir) or not os.path.isdir(basedir):
-            print("MTT_HOME points to an invalid location - please correct")
-            sys.exit(1)
-        topdir = os.path.join(os.environ['MTT_HOME'], "pylib")
-    except KeyError:
-        # didn't find MTT_HOME in the environment, so
-        # next see if $0 is an absolute path
-        if os.path.isabs(sys.argv[0]):
-            # try that location
-            path = os.path.dirname(sys.argv[0])
-            basedir = os.path.join(path, "pylib", "System")
-            topdir = os.path.join(path, "pylib")
-            if not os.path.exists(basedir) or not os.path.isdir(basedir):
-                print("A base directory for MTT was not specified, we do not appear")
-                print("to be in the MTT home directory, and MTT_HOME has not been")
-                print("set in the environment. We cannot continue as we will be")
-                print("unable to find the MTT libraries")
-                sys.exit(1)
-        else:
-            print("A base directory for MTT was not specified, we do not appear")
-            print("to be in the MTT home directory, and MTT_HOME has not been")
-            print("set in the environment. We cannot continue as we will be")
-            print("unable to find the MTT libraries")
-            sys.exit(1)
+# check to see if MTT_HOME exists
+if not os.path.exists(mtthome):
+    print("MTT_HOME points to a non-existent location:")
+    print("    ", mtthome)
+    print("Please correct")
+    sys.exit(1)
+
+# set topdir and check for existence
+topdir = os.path.join(mtthome, "pylib")
+if not os.path.exists(topdir) or not os.path.isdir(topdir):
+    print("MTT_HOME points to a location that does not\ninclude the \"pylib\" subdirectory:")
+    print("   ", topdir)
+    print("does not exist. Please correct")
+    sys.exit(1)
+
+# set basedir and check for existence
+basedir = os.path.join(mtthome, "pylib", "System")
```

Comment:
This is ignoring the fact that basedir might be set by `--base-dir`. Maybe this should be
```python
basedir = args.basedir or os.path.join(mtthome, "pylib", "System")
```

Reply:
okay, added some checks and supported use of base-dir